templates/datapipeline.yml

```diff
@@ -0,0 +1,309 @@
+    Type: AWS::EC2::Image::Id
+  myTopicArn:
+    Type: String
+  myS3LogBucket:
+    Type: String
+  myDataPipelineResourceRole:
+    Type: String
+  myDataPipelineRole:
+    Type: String
+  myKeyPair:
+    Type: AWS::EC2::KeyPair::KeyName
+  Tag:
+    Type: String
+  myPeriod:
+    Type: String
+
+
+Resources:
+  DataPipelineEFSBackup:
+    Type: AWS::DataPipeline::Pipeline
+    Properties:
+      Name: DataPipelineEFSBackup
+      Description: DataPipeline EFS Backup
+      PipelineTags:
+        - Key: Name
+          Value: !Ref Tag
+      PipelineObjects:
+        - Id: Default
+          Name: Default
+          Fields:
+            - Key: type
+              StringValue: Default
+            - Key: scheduleType
+              StringValue: cron
+            - Key: failureAndRerunMode
+              StringValue: CASCADE
+            - Key: schedule
+              RefValue: DefaultSchedule
+            - Key: role
+              StringValue: "#{myDataPipelineRole}"
+            - Key: resourceRole
+              StringValue: "#{myDataPipelineResourceRole}"
+            - Key: pipelineLogUri
+              StringValue: "#{myS3LogBucket}"
+        - Id: EC2ResourceObj
+          Name: EC2ResourceObj
+          Fields:
+            - Key: type
+              StringValue: Ec2Resource
+            - Key: terminateAfter
+              StringValue: "30 Minutes"
+            - Key: instanceType
+              StringValue: "#{myInstanceType}"
+            - Key: securityGroupIds
+              StringValue: "#{mySecurityGroupId}"
+            - Key: subnetId
+              StringValue: "#{mySubnetId}"
+            - Key: associatePublicIpAddress
+              StringValue: "true"
+            - Key: imageId
+              StringValue: "#{myImageId}"
+            - Key: keyPair
+              StringValue: "#{myKeyPair}"
+        - Id: DefaultSchedule
+          Name: DefaultSchedule
+          Fields:
+            - Key: type
+              StringValue: Schedule
+            - Key: startAt
+              StringValue: FIRST_ACTIVATION_DATE_TIME
+            - Key: period
+              StringValue: "#{myPeriod}"
+        - Id: ShellCommandActivityObj
+          Name: ShellCommandActivityObj
+          Fields:
+            - Key: type
+              StringValue: ShellCommandActivity
+            - Key: runsOn
+              RefValue: EC2ResourceObj
+            - Key: command
+              StringValue: |
+                efs_id="$1"
+                region="$2"
+                destination="$3"
+                source="${efs_id}.efs.${region}.amazonaws.com"
+                sudo yum -y install nfs-utils
+                [[ -d /backup ]] || sudo mkdir /backup
+                if ! mount -l -t nfs4 | grep -qF $source; then
+                  sudo mount -t nfs -o nfsvers=4.1 -o rsize=1048576 -o wsize=1048576 -o timeo=600 -o retrans=2 -o hard "$source":/ /backup
+                fi
+                sudo aws s3 sync /backup s3://$destination/${efs_id}/
```

Comment:
This should have the `--delete` flag so that it deletes files on S3 that do not exist locally. 
Also add `--exact-timestamps`.
I recommend when using `sync` operations on directories, to always use a `/` at the end. In this case it's not symmetrical.